src/python/pants/ivy/ivy_subsystem.py

```diff
@@ -40,6 +40,10 @@ def register_options(cls, register):
              help='The version of ivy to fetch.')
     register('--cache-dir', advanced=True, default=os.path.expanduser('~/.ivy2/pants'),
```

Comment:
Should deprecate this option probably?

Reply:
So, based on our discussion, I'm fine with not deprecating this. But you should expand the description of all of these options to explain how they interact, and you should add validation that both `cache-dir` and the more specific options aren't set at the same time (see my previous comments regarding `is_default`).